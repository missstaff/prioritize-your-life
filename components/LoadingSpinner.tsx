import { ActivityIndicator } from "react-native";
import { ScaledSheet, s } from "react-native-size-matters";
import { AppThemedView } from "./app_components/AppThemedView";
import { COLORS } from "@/constants/Colors";

type SizeType = number | "small" | "large";

interface LoadingSpinnerProps {
  color?: string;
  size?: SizeType;
}

/**
 * Renders a loading spinner component.
 * @param {LoadingSpinnerProps} props - The props for the LoadingSpinner component.
 * @param {string} props.size - The size of the spinner. Defaults to "large".
 * @param {string} props.color - The color of the spinner. Defaults to COLORS.primary.
 * @returns {JSX.Element} The rendered LoadingSpinner component.
 */
export default function LoadingSpinner({
  size = "large",
  color = COLORS.primary,
}: LoadingSpinnerProps): JSX.Element {
  return (
    <AppThemedView style={styles.container}>
      <ActivityIndicator size={size} color={color} />
    </AppThemedView>
  );
}

const styles = ScaledSheet.create({
  container: {
    alignItems: "center",
    display: "flex",
    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
  },
  horizontal: {
    flexDirection: "row",
    justifyContent: "space-around",
    padding: s(10),
  },
});
