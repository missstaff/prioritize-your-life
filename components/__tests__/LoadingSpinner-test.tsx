import * as React from "react";
import renderer from "react-test-renderer";
import LoadingSpinner from "../LoadingSpinner";

describe("LoadingSpinner", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  
  it("renders correctly with the given props", () => {
    const tree = renderer.create(<LoadingSpinner size="small" color="red" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});