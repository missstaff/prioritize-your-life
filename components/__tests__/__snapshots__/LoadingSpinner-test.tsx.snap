// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly with the given props 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#151718",
      },
      {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      },
    ]
  }
>
  <ActivityIndicator
    color="red"
    size="small"
  />
</View>
`;
